diesel_codegen/src/schema_inference.rs

```diff
@@ -11,19 +11,79 @@ pub fn derive_infer_schema(input: syn::MacroInput) -> quote::Tokens {
                with your invocation of `infer_schema!");
     }
 
-    let options = input.attrs.iter().find(|a| a.name() == "options").map(|a| &a.value);
-    let options = match options {
-        Some(&syn::MetaItem::List(_, ref options)) => options,
-        _ => bug(),
-    };
-    let database_url = options.iter().find(|a| a.name() == "database_url")
-        .map(|a| str_value_of_meta_item(a, "database_url"))
-        .unwrap_or_else(|| bug());
+    let options = get_options_from_input(&input.attrs, bug);
+    let database_url = get_option(&options, "database_url", bug);
 
     let table_names = load_table_names(&database_url).unwrap();
     let schema_inferences = table_names.into_iter().map(|table_name| {
         quote!(infer_table_from_schema!(#database_url, #table_name);)
```

Comment:
I love how this tiny line right here is sets of a huge awesome piece of rust-macro-derive-inception as it (1) is part of a macro (2) that is contained a custom derive (3) which expands to a (set of) macro calls (4) that themselves include custom derives (5) where each one actually expands to calls to the `tables!` macros. You could probably write a few pages of internal docs just about that 😄